File: caseStudy_1/Java_programs/Count_trailing_zeroes.java
package Java_programs;
import java.util.*;
class count_zeros{
    public static void trailing_zeros(int n){
        int count = 0;
        while(n>=5){
            n=n/5;
            count +=n;
        }
        System.out.println(count);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("enter value to find factorial:-");
        int n= sc.nextInt();
        trailing_zeros(n);
        sc.close();
    }
}

File: caseStudy_1/Python_programs/Count_trailing_zeroes.py

# ---------- efficient approach ------------

# def count_Zeros(n):
#     count = 0
#     while(n>=5):
#         n=n//5
#         count+=n
#     return count


# n = int(input("enter n value"))
# print(count_Zeros(n))

# ------------ 2nd approach -------------------
import math
n = 100
factorial = math.factorial(n)
print(factorial)
count=0
fact_string = str(factorial)
for char in reversed(fact_string):
    if char =='0':
        count+=1
    else:
        break
print(count)